source/elements/oneDPL/source/parallel_api.rst

```diff
@@ -688,5 +688,41 @@ than an element in the range being searched.
 
 The elements e of [start, end) must be partitioned with respect to the comparator used.
 
+.. code:: cpp
+
+  template <typename Policy, typename InputIt, typename OutputIt, typename UnaryOp,
+      typename UnaryPredicate>
+  OutputIt
+  transform_if(Policy&& policy, InputIt start, InputIt end, OutputIt result, UnaryOp op,
+      UnaryPredicate pred);                                                               // (1)
+
+  template <typename Policy, typename InputIt1, typename InputIt2, typename OutputIt, 
+      typename BinaryOp, typename BinaryPredicate>
+  OutputIt
+  transform_if(Policy&& policy, InputIt1 start1, InputIt1 end1, InputIt2 start2, OutputIt result,
+      BinaryOp op, BinaryPredicate pred);                                                 // (2)
+
+:code:`oneapi::dpl::transform_if` applies a given function to the elements of the input sequence(s) that
+satisfy a given predicate, and stores the result to the output. Depending on the arguments, the algorithm
+
+1.
+evaluates the unary predicate :code:`pred` for each position :code:`i` of the sequence
+:code:`[start, end)` and if :code:`pred(start[i]) == true`, it performs the unary operation
+:code:`op(start[i])` and stores the result into :code:`result[i]`. If
+:code:`pred(start[i]) == false`, the data element :code:`result[i]` is not modified from its
+initial value. The return is an iterator to the element of the output sequence
+:code:`result + std::distance(start,end)`, one past the last element which was considered
+for modification by the algorithm.
```

Comment:
```suggestion
initial value. The return value is an iterator targeting past the last considered element
of the output sequence, that is, :code:`result + (start - end)`.
```
Alternatively, we can use `std::advance(result, std::distance(start, end))` - but all sequences are random access anyway, so that seems excessive.

Reply:
I agree subtraction is better (but with `end - start`).  fixed.